internal/printer/fake/fake.go

```diff
@@ -16,23 +16,45 @@ package fake
 
 import (
 	"context"
+	"io"
+	"os"
 
 	"github.com/GoogleContainerTools/kpt/internal/pkg"
 	"github.com/GoogleContainerTools/kpt/internal/printer"
 )
 
-// NilPrinter implements the printer.Printer interface and just ignores
+// Printer implements the printer.Printer interface and just ignores
 // all print calls.
-type NilPrinter struct{}
+type Printer struct {
+	outStream io.Writer
+	errStream io.Writer
+}
+
+func (np *Printer) PrintPackage(*pkg.Pkg, bool) {}
+
+func (np *Printer) OptPrintf(*printer.Options, string, ...interface{}) {}
+
+func (np *Printer) Printf(string, ...interface{}) {}
 
-func (np *NilPrinter) PrintPackage(*pkg.Pkg, bool) {}
+func (np *Printer) OutStream() io.Writer { return np.outStream }
 
-func (np *NilPrinter) OptPrintf(*printer.Options, string, ...interface{}) {}
+func (np *Printer) ErrStream() io.Writer { return np.errStream }
 
-func (np *NilPrinter) Printf(string, ...interface{}) {}
+// CtxWithEmptyPrinter returns a new context with the Empty printer added.
+func CtxWithEmptyPrinter() context.Context {
+	return CtxWithPrinter(nil, nil)
+}
+
+// CtxWithDefaultPrinter returns a new context with the printer which has os streams
+func CtxWithDefaultPrinter() context.Context {
+	return CtxWithPrinter(os.Stdout, os.Stderr)
```

Comment:
I usually prefer to just discard output from tests instead of sending it to the console. It just tends to make the test output hard to read.